proposals/pattern-const-expressions.md

```diff
@@ -111,9 +111,54 @@ const bool z = d is not null; // always true
 const bool p = b is null; // always false
 ```
 
-All the above are currently valid pattern matching expressions, that also emit warnings about their constant evaluation results, about them being always true or false.
+### Diagnostics
 
-When assigning those expressions to a constant symbol, these warnings about the constant result of the expression will **not** be reported, as the user intends to capture the constant value of the expression.
+All the above are currently valid pattern matching expressions, that also emit warnings about their constant evaluation results, being always true or false.
+
+When assigning those expressions in a constant context, these warnings about the constant result of the expression will **not** be reported, as the user intends to capture the constant value of the expression. Constant context involves:
+- initializers for const symbols (fields or locals)
+- attribute arguments
+- parameter default value
+- switch statement case labels
+- switch expression arms
```

Comment:
We need to see the actual spec changes here. How do we have to redefine the sections that specify these constructs to introduce a constant context? What is the definition of a constant context itself?